control-plane/agents/src/bin/core/volume/clone_operations.rs

```diff
@@ -136,19 +137,24 @@ impl CreateVolumeExe for SnapshotCloneOp<'_> {
     async fn create<'a>(
         &'a self,
         context: &mut Context<'a>,
-        clone_replica: Self::Candidates,
+        clone_replicas: Self::Candidates,
     ) -> Vec<Replica> {
-        match OperationGuardArc::<ReplicaSpec>::create_ext(context.registry, &clone_replica).await {
-            Ok(replica) => vec![replica],
-            Err(error) => {
-                context.volume.error(&format!(
-                    "Failed to create replica {:?} for volume, error: {}",
-                    clone_replica,
-                    error.full_string()
-                ));
-                vec![]
+        let mut replicas = vec![];
+        for clone_replica in &clone_replicas {
+            match OperationGuardArc::<ReplicaSpec>::create_ext(context.registry, clone_replica)
+                .await
+            {
+                Ok(replica) => replicas.push(replica),
+                Err(error) => {
+                    context.volume.error(&format!(
```

Comment:
What happens on failure, should we attempt to create up to N replicas and break out as soon as we got N ?

Reply:
1. if replication factor for the new clone volume is not same with source snapshots, the We will reject the operation. `InsufficientSnapshotsForClone`
2. If replica creation failed from one of the snapshot, it will follow the base logic.

Reply:
if clone replicas > source replicas, then we can still create, with rebuild, but this can be discussed later.
but if clone replicas < source replicas, then we can still create right?

Reply:
Indeeed yes, if clone replicas < source replicas, then we should make it success. I will change it.